lib/Bio/EnsEMBL/DataCheck/Checks/CheckTableSizes.pm

```diff
@@ -0,0 +1,95 @@
+=head1 LICENSE
+
+Copyright [2018-2019] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::CheckTableSizes;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+use Bio::EnsEMBL::DataCheck::Utils qw/ array_diff /;
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME           => 'CheckTableSizes',
+  DESCRIPTION    => 'Tables must be populated and not differ significantly in row numbers',
+  GROUPS         => ['compara', 'compara_tables'],
+  DATACHECK_TYPE => 'advisory',
+  DB_TYPES       => ['compara'],
+};
+
+sub tests {
+  my ($self) = @_;
+  
+  my $curr_dba = $self->dba;
+  my $curr_helper = $curr_dba->dbc->sql_helper;
+  my $prev_dba = $self->registry->get_DBAdaptor('compara_prev', 'compara') || $self->get_old_dba;
+  my $prev_helper = $prev_dba->dbc->sql_helper;
+  
+  my $table_sql = "SHOW TABLES";
+  my $curr_tables = $curr_helper->execute_simple( -SQL => $table_sql );
+  my $prev_tables = $prev_helper->execute_simple( -SQL => $table_sql );
+  my $curr_db_name = $curr_dba->dbc->dbname;
+  my $prev_db_name = $prev_dba->dbc->dbname;
+  
+  my %prev_tables = ();
+  foreach my $table ( @$prev_tables ) {
+    $prev_tables{$table}++;
+  }
+  my %curr_tables = ();
+  foreach my $table ( @$curr_tables ) {
+    $curr_tables{$table}++;
+  }
+  
+  foreach my $table ( @$curr_tables ) {
+    my $desc_1 = "The number of rows in $table for $curr_db_name has not increased by >10% from $prev_db_name";
+    my $desc_2 = "The number of rows in $table for $curr_db_name has not decreased by >5% from $prev_db_name";
+    my $desc_3 = "The number of rows in $table has not remained static between $curr_db_name and $prev_db_name";
+    my $sql = qq/
+      SELECT COUNT(*) FROM $table
+    /;
+      
+    if ( exists($prev_tables{$table}) ) {
+      my $prev_row_count = $prev_helper->execute_single_result( -SQL => $sql );
+      my $curr_row_count = $curr_helper->execute_single_result( -SQL => $sql );
+
+      cmp_ok( $curr_row_count, '<=', ($prev_row_count*1.1), $desc_1 );
```

Comment:
Note that there is a method in Bio::EnsEMBL::DataCheck::Test::DataCheck, `row_totals` which wraps this functionality up and enables a bit more succinctness. Admittedly, you would be doing the count multiple times, rather than using the count variable - but we can probably assume that mysql caching will mitigate that. `row_totals` is also perhaps a bit less readable, so I can see why you wouldn't want to use it, but just wanted to make sure you knew it was there...

Reply:
Thank you, I did use `row_totals` in another version of this CheckTableSizes, but decided on this way for the reasons you have mentioned. I think I have used it in another DC however.